lib/src/metta/runner/stdlib_math.rs

```diff
@@ -0,0 +1,730 @@
+use crate::*;
+use crate::metta::*;
+#[cfg(feature = "pkg_mgmt")]
+
+use std::convert::TryInto;
+
+use super::arithmetics::*;
+
+
+macro_rules! grounded_op {
+    ($name:ident, $disp:literal) => {
+        impl PartialEq for $name {
+            fn eq(&self, _other: &Self) -> bool {
+                true
+            }
+        }
+
+        impl std::fmt::Display for $name {
+            fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+                write!(f, $disp)
+            }
+        }
+    }
+}
+
+#[derive(Clone, Debug)]
+pub struct PowiMathOp {}
+
+grounded_op!(PowiMathOp, "powi-math");
+
+impl Grounded for PowiMathOp {
+    fn type_(&self) -> Atom {
+        Atom::expr([ARROW_SYMBOL, ATOM_TYPE_NUMBER, ATOM_TYPE_NUMBER, ATOM_TYPE_NUMBER])
+    }
+
+    fn as_execute(&self) -> Option<&dyn CustomExecute> {
+        Some(self)
+    }
+}
+
+impl CustomExecute for PowiMathOp {
+    fn execute(&self, args: &[Atom]) -> Result<Vec<Atom>, ExecError> {
+        let arg_error = || ExecError::from("powi-math expects two arguments: number (base) and number (power)");
+        let base = Into::<f64>::into(args.get(0).and_then(Number::from_atom).ok_or_else(arg_error)?);
+        let pow = Into::<i64>::into(args.get(1).and_then(Number::from_atom).ok_or_else(arg_error)?);
+        let res = base.powi(pow.try_into().unwrap());
+        Ok(vec![Atom::gnd(Number::Float(res))])
+    }
+}
+
+#[derive(Clone, Debug)]
+pub struct PowfMathOp {}
+
+grounded_op!(PowfMathOp, "powf-math");
+
+impl Grounded for PowfMathOp {
+    fn type_(&self) -> Atom {
+        Atom::expr([ARROW_SYMBOL, ATOM_TYPE_NUMBER, ATOM_TYPE_NUMBER, ATOM_TYPE_NUMBER])
+    }
+
+    fn as_execute(&self) -> Option<&dyn CustomExecute> {
+        Some(self)
+    }
+}
+
+impl CustomExecute for PowfMathOp {
+    fn execute(&self, args: &[Atom]) -> Result<Vec<Atom>, ExecError> {
+        let arg_error = || ExecError::from("powf-math expects two arguments: number (base) and number (power)");
+        let base = Into::<f64>::into(args.get(0).and_then(Number::from_atom).ok_or_else(arg_error)?);
+        let pow = Into::<f64>::into(args.get(1).and_then(Number::from_atom).ok_or_else(arg_error)?);
+        let res = base.powf(pow.try_into().unwrap());
+        Ok(vec![Atom::gnd(Number::Float(res))])
+    }
+}
+
+#[derive(Clone, Debug)]
+pub struct SqrtMathOp {}
+
+grounded_op!(SqrtMathOp, "sqrt-math");
+
+impl Grounded for SqrtMathOp {
+    fn type_(&self) -> Atom {
+        Atom::expr([ARROW_SYMBOL, ATOM_TYPE_NUMBER, ATOM_TYPE_NUMBER])
+    }
+
+    fn as_execute(&self) -> Option<&dyn CustomExecute> {
+        Some(self)
+    }
+}
+
+impl CustomExecute for SqrtMathOp {
+    fn execute(&self, args: &[Atom]) -> Result<Vec<Atom>, ExecError> {
+        let arg_error = || ExecError::from("sqrt-math expects one argument: number");
+        let input = Into::<f64>::into(args.get(0).and_then(Number::from_atom).ok_or_else(arg_error)?);
```

Comment:
Just to make code a little bit more readable you can use here and  below:
```suggestion
        let input: f64 = args.get(0).and_then(Number::from_atom).ok_or_else(arg_error)?.into();
```